pkg/common/event/dml_event.go

```diff
@@ -349,17 +349,21 @@ func (t *DMLEvent) AppendRow(raw *common.RawKVEntry,
 	if err != nil {
 		return err
 	}
-	for range count {
-		t.RowTypes = append(t.RowTypes, rowType)
+	if count > 0 {
+		for range count {
+			t.RowTypes = append(t.RowTypes, rowType)
 
-		keyCopy := make([]byte, len(raw.Key))
-		copy(keyCopy, raw.Key)
-		t.RowKeys = append(t.RowKeys, keyCopy)
-	}
-	t.Length += 1
-	t.ApproximateSize += raw.GetSize()
-	if checksum != nil {
-		t.Checksum = append(t.Checksum, checksum)
+			keyCopy := make([]byte, len(raw.Key))
+			copy(keyCopy, raw.Key)
+			t.RowKeys = append(t.RowKeys, keyCopy)
+		}
+		t.Length += 1
+		t.ApproximateSize += raw.GetSize()
+		if checksum != nil {
+			t.Checksum = append(t.Checksum, checksum)
+		}
+	} else {
+		log.Panic("DMLEvent.AppendRow: no rows decoded from the raw KV entry", zap.Any("raw", raw))
```

Comment:
panic immediate if count == 0, remove the else

Reply:
done